### PR TITLE
Convert userLibraryAccessPage to react

### PR DESCRIPTION
**Pull request by grafixeyehero**
Depend on   Convert edit user page to react #3070

### CODE REVIEW

File: src/components/pages/UserLibraryAccessPage.tsx [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+        loadChannels(user, channels);
+        loadMediaFolders(user, mediaFolders);
+        loadDevices(user, devices);
+        loading.hide();
+    }, [loadChannels, loadDevices, loadMediaFolders]);
+
+    const loadData = useCallback(() => {
+        loading.show();
+        const userId = appRouter.param('userId');
+        // eslint-disable-next-line compat/compat
```
Comment (thornbill) [inline src/components/pages/UserLibraryAccessPage.tsx:123]: Is this required to use `Promise`?

File: src/components/pages/UserLibraryAccessPage.tsx [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+            loading.hide();
+            toast(globalize.translate('SettingsSaved'));
+        };
+
+        element?.current?.querySelector('.chkEnableAllDevices').addEventListener('change', function (this: HTMLInputElement) {
+            if (this.checked) {
+                element?.current?.querySelector('.deviceAccessListContainer').classList.add('hide');
+            } else {
+                element?.current?.querySelector('.deviceAccessListContainer').classList.remove('hide');
+            }
```
Comment (thornbill) [inline src/components/pages/UserLibraryAccessPage.tsx:186]: It would probably be better to use `classList.toggle()` for these.

**Review by thornbill — CHANGES_REQUESTED** [review]
Looks great overall! I just had a couple minor questions/comments.

Should the console.log statements be removed?

File: src/components/pages/UserLibraryAccessPage.tsx (hunk 2 above)
Comment (grafixeyehero) [reply src/components/pages/UserLibraryAccessPage.tsx]: cf39bc0

File: src/components/pages/UserLibraryAccessPage.tsx (hunk 1 above)
Comment (grafixeyehero) [reply src/components/pages/UserLibraryAccessPage.tsx:120]: previously was used  jQuery $.deferred
https://github.com/jellyfin/jellyfin-web/blob/7ea8ce7debae59cb91c8f1f3a647931f49217c93/src/controllers/dashboard/users/userlibraryaccess.js#L165-L173

Comment (thornbill) [reply src/components/pages/UserLibraryAccessPage.tsx:120]: Ah I see this is an issue with our eslint configuration. The polyfills are not being included for jsx/tsx files. That probably deserves a separate PR. :smile:

**Review by thornbill — APPROVED** [review]

**Pull request merged**